pkg/sql/opt/norm/testdata/rules/join

```diff
@@ -1095,6 +1098,149 @@ inner-join (hash)
  └── filters
       └── d.x:11 = xy.x:14 [outer=(11,14), constraints=(/11: (/NULL - ]; /14: (/NULL - ]), fd=(11)==(14), (14)==(11)]
 
+# Regression test for #46151. Do not push down a filter with a correlated
+# subquery.
+norm expect-not=PushFilterIntoJoinLeftAndRight
```

Comment:
Can this test be further simplified? These huge SQL-Smith style queries are a real pain to deal with when plans change.

Reply:
Done.